test/server/cards/01_SOR/events/TheEmperorsLegion.spec.ts

```diff
@@ -0,0 +1,55 @@
+describe('The Emprerors Legion', function () {
+    integration(function (contextRef) {
+        it('should return card to player hand from a discard pile', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['the-emperors-legion'],
+                    groundArena: ['pyke-sentinel', 'seasoned-shoretrooper', 'blizzard-assault-atat'],
+                    spaceArena: ['cartel-spacer'],
+                    discard: ['keep-fighting', 'green-squadron-awing', 'disarm']
+                },
+                player2: {
+                    hand: ['superlaser-blast'],
+                    discard: ['tactical-advantage', 'guerilla-attack-pod']
+                }
+            });
+            const { context } = contextRef;
+
+            context.player1.passAction();
+            context.player2.clickCard(context.superlaserBlast);
+            expect(context.pykeSentinel.location).toBe('discard');
+            expect(context.seasonedShoretrooper.location).toBe('discard');
+            expect(context.blizzardAssaultAtat.location).toBe('discard');
+            expect(context.cartelSpacer.location).toBe('discard');
+            context.player1.clickCard(context.theEmperorsLegion);
+            expect(context.pykeSentinel.location).toBe('hand');
+            expect(context.seasonedShoretrooper.location).toBe('hand');
+            expect(context.blizzardAssaultAtat.location).toBe('hand');
+            expect(context.cartelSpacer.location).toBe('hand');
+            expect(context.player2).toBeActivePlayer();
+            // Expect earlier defeated units to still be in the discard pile
```

Comment:
Actually we may need to set this test up slightly differently. When the test harness populates the discard pile, it kind of cheats the cards in so they were never actually "defeated" from a game perspective.

Ideally we would want to test two phases where we defeat some cards explicitly in the first phase using attacks or something, use `context.moveToNextActionPhase()` to move to a second phase, then do this test with SLB.

We should do something similar for the below test except in the second phase we don't defeat any units.